packages/beeq/src/components/tag/bq-tag.tsx

```diff
@@ -0,0 +1,221 @@
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /** Method to be called to remove the tag component */
+  @Method()
+  async hide(): Promise<void> {
+    this.handleHide();
+  }
+
+  /** Method to be called to show the alert component */
+  @Method()
+  async show(): Promise<void> {
+    this.handleShow();
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleHide = () => {
+    const ev = this.bqHide.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = false;
+    }
+  };
+
+  private handleShow = () => {
+    const ev = this.bqShow.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = true;
+    }
+  };
+
+  private handleClick = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    const ev = this.bqClick.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.selected = !this.selected;
+    }
+  };
+
+  private handleFocus = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    this.bqFocus.emit(this.el);
+  };
+
+  private get iconSize(): number {
+    return SIZE_TO_VALUE_MAP[this.size] || SIZE_TO_VALUE_MAP.medium;
+  }
+
+  private handleOnKeyDown = (event: KeyboardEvent) => {
+    this.bqKeyDown.emit(event);
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <Host
+        class={{ 'is-hidden': !this.open }}
+        aria-hidden={!this.open ? 'true' : 'false'}
+        hidden={!this.open ? 'true' : 'false'}
+      >
+        <div
+          class={{
+            'bq-tag': true,
+            disabled: this.disabled,
+            active: !this.disabled && this.selected,
+            [`bq-tag__wrapper--${this.size} font-medium leading-regular`]: true,
+            [`bq-tag__${this.type}__${this.variant}`]: this.hasColor,
+          }}
+          aria-selected={this.selected}
+          aria-disabled={this.disabled}
+          onClick={this.handleClick}
+          onFocus={this.handleFocus}
+          onKeyDown={this.handleOnKeyDown}
+          role="tab"
+          part="wrapper"
+          tabindex={this.disabled ? '-1' : '0'}
+        >
+          <div class={{ 'bq-tag__icon': true, '!hidden': !this.hasIcon }}>
+            <slot name="icon">
+              <bq-icon size={this.iconSize} name="star" part="icon" exportparts="base,svg" />
+            </slot>
+          </div>
+          <div
+            class={{
+              'text-xs': this.size === 'extra_small',
+              'text-s': this.size === 'small',
+              'text-m': this.size === 'medium',
+            }}
+          >
```

Comment:
We are missing exporting the shadow DOM part here:

```suggestion
          <div
            class={{
              'text-xs': this.size === 'extra_small',
              'text-s': this.size === 'small',
              'text-m': this.size === 'medium',
            }}
            part="text"
          >
```